app/javascript/dashboard/mixins/automations/methodsMixin.js

```diff
@@ -0,0 +1,311 @@
+      return this.automationTypes[key].conditions;
+    },
+    getInputType(key) {
+      const customAttribute = isACustomAttribute(this.allCustomAttributes, key);
+      if (customAttribute) {
+        return getCustomAttributeInputType(
+          customAttribute.attribute_display_type
+        );
+      }
+      const type = this.getAutomationType(key);
+      return type.inputType;
+    },
+    getOperators(key) {
+      if (this.mode === 'edit') {
+        const customAttribute = isACustomAttribute(
+          this.allCustomAttributes,
+          key
+        );
+        if (customAttribute) {
+          return getOperatorTypes(customAttribute.attribute_display_type);
+        }
+      }
+      const type = this.getAutomationType(key);
+      return type.filterOperators;
+    },
+    getAutomationType(key) {
+      return this.automationTypes[this.automation.event_name].conditions.find(
+        condition => condition.key === key
+      );
+    },
+
+    getConditionDropdownValues(type) {
+      const {
+        agents,
+        allCustomAttributes: customAttributes,
+        booleanFilterOptions,
+        campaigns,
+        contacts,
+        inboxes,
+        labels,
+        statusFilterOptions,
+        teams,
+      } = this;
+      return getConditionOptions({
+        agents,
+        booleanFilterOptions,
+        campaigns,
+        contacts,
+        customAttributes,
+        inboxes,
+        labels,
+        statusFilterOptions,
+        teams,
+        type,
+      });
+    },
+    appendNewCondition() {
+      if (this.mode === 'edit') {
+        if (
+          !this.automation.conditions[this.automation.conditions.length - 1]
+            .query_operator
+        ) {
+          this.automation.conditions[
+            this.automation.conditions.length - 1
+          ].query_operator = 'and';
+        }
+      }
+      this.automation.conditions.push(
+        ...getDefaultConditions(this.automation.event_name)
+      );
+    },
+    appendNewAction() {
+      this.automation.actions.push(...getDefaultActions());
+    },
+    removeFilter(index) {
+      if (this.automation.conditions.length <= 1) {
+        this.showAlert(this.$t('AUTOMATION.CONDITION.DELETE_MESSAGE'));
+      } else {
+        this.automation.conditions.splice(index, 1);
+      }
+    },
+    removeAction(index) {
+      if (this.automation.actions.length <= 1) {
+        this.showAlert(this.$t('AUTOMATION.ACTION.DELETE_MESSAGE'));
+      } else {
+        this.automation.actions.splice(index, 1);
+      }
+    },
+    submitAutomation() {
+      this.$v.$touch();
+      if (this.$v.$invalid) return;
+      const automation = JSON.parse(JSON.stringify(this.automation));
+      automation.conditions[
+        automation.conditions.length - 1
+      ].query_operator = null;
+      automation.conditions = filterQueryGenerator(
+        automation.conditions
+      ).payload;
+      automation.actions = actionQueryGenerator(automation.actions);
```

Comment:
This can be a helper method, return a new automation object.

Reply:
Yes, I have made some more changes to this, pushing the code in a bit @pranavrajs